spring-webflux/src/main/java/org/springframework/web/reactive/socket/server/support/HandshakeWebSocketService.java

```diff
@@ -272,7 +272,8 @@ private HandshakeInfo createHandshakeInfo(ServerWebExchange exchange, ServerHttp
 			@Nullable String protocol, Map<String, Object> attributes) {
 
 		URI uri = request.getURI();
-		HttpHeaders headers = request.getHeaders();
+		HttpHeaders headers = new HttpHeaders();
+		headers.addAll(request.getHeaders());
```

Comment:
As mentioned in the commit message, this is the only place where we actually need to clone the request headers.
Because native headers are often pooled by the server, those values are recycled by the time the HTTP handshake is done and those aren't available anymore during the websocket session.